ironfish/src/rpc/routes/wallet/createAccount.ts

```diff
@@ -45,10 +59,23 @@ routes.register<typeof CreateAccountRequestSchema, CreateAccountResponse>(
   CreateAccountRequestSchema,
   async (request, context): Promise<void> => {
     AssertHasRpcContext(request, context, 'wallet')
+    if (
+      request.data.createdAt?.hash === undefined ||
+      request.data.createdAt?.sequence === undefined
+    ) {
+      request.data.createdAt = undefined
+    }
+
+    const createdAt = request.data.createdAt && {
+      hash: Buffer.from(request.data.createdAt.hash, 'hex'),
+      sequence: request.data.createdAt.sequence,
+    }
+
+    const setDefault = !context.wallet.hasDefaultAccount || (request.data.default ?? false)
```

Comment:
random nit: but I prefer small refactors like this split out to other PRs. This is not related to the PR.

Reply:
Also because now that this code changed, the code is actually very strange that the RPC is auto setting the default account if one is not set when an account is created. This should absolutely not be happening and should be removed in another PR.

Reply:
I can move this refactor to another PR. I'm still confused on how it changes the functionality though. In the previous code if there was no default account before this RPC call, it would set this account as the default. Seems the same as what this code is doing

Reply:
There's no change in functionality here. I re-read previous comments and don't see any mention of that. That's the issue I'm mentioning, it's a refactor that doesn't change functionality for an orthagonal piece of code unrelated to this change. (using the internal default setting rather than duplicating it)